### ISSUE TITLE
examples/widgets/span_selector.py is brittle

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#20200]
### Problem

examples/widgets/span_selector.py crashes if one clicks somewhere and immediately releases the click, without dragging the SpanSelector, with
```
Traceback (most recent call last):
  File ".../matplotlib/lib/matplotlib/cbook/__init__.py", line 274, in process
    func(*args, **kwargs)
  File ".../matplotlib/lib/matplotlib/widgets.py", line 1840, in release
    self._release(event)
  File ".../matplotlib/lib/matplotlib/widgets.py", line 2069, in _release
    self.onselect(vmin, vmax)
  File ".../matplotlib/examples/widgets/span_selector.py", line 36, in onselect
    ax2.set_xlim(region_x[0], region_x[-1])
IndexError: index 0 is out of bounds for axis 0 with size 0
```

### Suggested Improvement

Make the example more robust.

**Matplotlib version**

  * Operating system: linux
  * Matplotlib version (`import matplotlib; print(matplotlib.__version__)`): 3.4+HEAD
  * Matplotlib documentation version (is listed under the logo): 3.4.2.post693

**Comment by Jashshah6086** [comment]
I would like to work on it.

**Comment by anntzer** [comment]
It's yours.

**Comment by Jashshah6086** [comment]
https://github.com/matplotlib/matplotlib/pull/20212#issue-643122459

- Why Error occured?
when pointer is not dragged list is either null or has null element.Which created error in indexing. 

- Solution:
Added condition on length of region_x 

```
if len(region_x) >= 2:
        line2.set_data(region_x, region_y)
        ax2.set_xlim(region_x[0], region_x[-1])
        ax2.set_ylim(region_y.min(), region_y.max())
        fig.canvas.draw()
```
instead of

```
line2.set_data(region_x, region_y)
ax2.set_xlim(region_x[0], region_x[-1])
ax2.set_ylim(region_y.min(), region_y.max())
fig.canvas.draw()
```

Graph with less than 2 points makes no sense. Hence, after change there is no change in plot and no error in command line when not dragged.

**Comment by QuLogic** [comment]
Closed by #20212.
